lib/mage/Mage.js

```diff
@@ -262,19 +263,52 @@ Mage.prototype.setupCoreLibs = function () {
 	core.cmd = require('../commandCenter');
 	core.sampler = require('../sampler');
 
+	// Setup a core messenger system for passing messages between
+	// core systems
+	core.processMessenger = new core.ProcessMessenger('mage.core');
+	core.processMessenger.on('exit', function (message) {
+		that.exit(message.exitCode, message.hard);
+	});
+
 	// Make this chainable.
 	return this;
 };
 
+/**
+ * Close down the MAGE server
+ *
+ * Workers will forward the request to their master if needed.
+ * This differs from `exit` since exit is equivalent to `process.exit`;
+ * see `exit` for more details.
+ *
+ * @param {number}  exitCode An exit code for node to return.
+ * @param {boolean} hard Wait for all services to closes before exiting
```

Comment:
.. if false

Reply:
You mean if true?

Reply:
`hard` being true sounds to me like it *wouldn't* wait, right?

Reply:
.... ah. These were copy/pasted from the current doc, wasn't paying attention.